libraries/apollo-mockserver/src/commonMain/kotlin/com/apollographql/apollo3/mockserver/http.kt

```diff
@@ -74,18 +85,55 @@ internal fun readRequest(source: BufferedSource): MockRequest? {
     "Transfer-Encoding $transferEncoding is not supported"
   }
 
-  val buffer = Buffer()
-  if (contentLength > 0) {
-    source.read(buffer, contentLength)
+  val body = if (contentLength > 0) {
+    readBytes(contentLength)
   } else if (transferEncoding == "chunked") {
-    source.readChunked(buffer)
+    val buffer2 = Buffer()
+    /**
+     * Read a source encoded in the "Transfer-Encoding: chunked" encoding.
+     * This format is a sequence of:
+     * - chunk-size (in hexadecimal) + CRLF
+     * - chunk-data + CRLF
+     */
+    while (true) {
+      val chunkSize = nextLine().trimEol().toLong(16)
+      if (chunkSize == 0L) {
+        check(nextLine() == "\r\n") // CRLF
+        break
+      } else {
+        buffer2.writeAll(readBytes(chunkSize))
+        check(nextLine() == "\r\n") // CRLF
+      }
+    }
+    buffer2
+  } else {
+    Buffer()
   }
 
+
   return MockRequest(
       method = method,
       path = path,
       version = version,
       headers = headers,
-      body = buffer.readByteString()
+      body = body.readByteString()
   )
 }
+
+private fun String.trimEol() = this.trimEnd('\r', '\n')
+
+internal suspend fun writeResponse(response: MockResponse, version: String, writeData: WriteData) {
+  writeData("$version ${response.statusCode}\r\n".encodeToByteArray())
+
+  // We don't support 'Connection: Keep-Alive', so indicate it to the client
+  val headers = response.headers //+ mapOf("Connection" to "close")
```

Comment:
Should this be commented?

Reply:
It should as we now support keep-alive 🎉 . But those 2 comments are very much inconsistent. Will fix.